### ISSUE TITLE
Ability to set log levels

### ISSUE THREAD

**Issue by friendofasquid** [dbt-labs/dbt-core#2385]
### Describe the feature

Allow configuring the log level, e.g. from `INFO` to `WARN` or `ERROR`. Could be configurable as a global CLI flag.

### Additional context

Some of our SQL occasionally has sensitive information that we wouldn't like to be logged. 

### Who will this benefit?

- People who want less "chatty" logging
- People using Spark external tables with credentials.

**Comment by drewbanin** [comment]
Thanks for opening this issue @friendofasquid. My instinct is that rather than configure log levels, we'd want to support masking sensitive values in the logs.

I think we can do this by:
 - Automatically masking sensitive values defined in the active `target` (passwords, etc)
 - Adding a context function that lets you register new sensitive values, eg:

```
select
  '{{ sensitive(env_var('MY_SENSITIVE_VAR')) }}'
...
```

Here, the `sensitive` context function would add a value to a list of sensitive values to mask in the log handler. This isn't a spec per se - there are likely other good ways to go about adding masking like this, and I don't have strong instincts about best practices around logging like this.

@beckjake do you have any strong opinions on how something like this could work?

**Comment by beckjake** [comment]
I think it's hard to say without knowing more about what spark external tables with credentials look like - does querying them include the credentials in the `select` statement, or something?

It seems like this user's needs could potentially be served by adding a flag to only log to the file at INFO level, which seems like a very reasonable flag to me.

I think the sensitive functionality would have to be limited to text-mode logging, as digging through the extras dict for sensitive data sounds tedious and error-prone. I'm not sure on the performance impact (my guess is: more than you'd think!) but you could just do something along the lines of:
```
for replacement in GLOBAL_REPLACEMENTS:
  msg = msg.replace(replacement, '*' * len(replacement))
```

or use a huge regex pattern by combining all of the replacements and call `re.sub` - that might be faster, I'd have to test to be sure.

The `'*' * len(replacement)` replacement scheme leaks information, but it doesn't screw up the formatting... trade-offs.

**Comment by friendofasquid** [comment]
Here's what a Spark external table declaration might look like, within a simple dbt template:

{% statement() %}
CREATE TEMPORARY VIEW source
USING jdbc
OPTIONS (
  url 'jdbc:postgresql://127.0.0.1/d?',
  user 'scott',
  password 'password',
  dbtable 't',
  driver 'org.postgresql.Driver'
);
{% endstatement %}

SELECT * FROM source

**Comment by rajeshmese** [comment]
@friendofasquid @beckjake is there any way to hide or mask the password getting logged in to dbt logs?

**Comment by friendofasquid** [comment]
@rajeshmese — I'm not aware of any enhancements that would help, unfortunately. We still struggle with this.

**Comment by Gunnnn** [comment]
Hello! Please reopen this issue cause i have trouble with thousands of rows in my dbt log and output from Databricks when having mistake and can not reduce it somehow!

**Comment by jtcohen6** [comment]
@Gunnnn That sounds like it may be caused by something unrelated. Shot in the dark: Any chance you're using python 3.10? It sounds like folks are seeing tons of extra log lines related to some deprecation warnings in a dependency (https://github.com/dbt-labs/dbt-core/issues/4537), something we plan to fix when we add official support (https://github.com/dbt-labs/dbt-core/issues/4562)

**Comment by Gunnnn** [comment]
> @Gunnnn That sounds like it may be caused by something unrelated. Shot in the dark: Any chance you're using python 3.10? It sounds like folks are seeing tons of extra log lines related to some deprecation warnings in a dependency (#4537), something we plan to fix when we add official support (#4562)

dbt version: 1.0.1
python version: 3.7.9
I see a lot of strings from Databricks stacktrace.
